textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexToGramQueryTranslatorTest.java

```diff
@@ -17,133 +17,321 @@ private void printTranslatorResult(String regex) {
+		
+		System.out.println("Simplified DNF: ");
+		System.out.println(simplifiedDNF.printQueryTree());
+
 		System.out.println();
 	}
 	
 	@Test
 	public void testEmptyRegex() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("");
+		String regex = "";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
 		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.ANY);
+		
+//		printTranslatorResult(regex);
 
-		Assert.assertTrue(exactQuery.equals(expectedQuery));
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
 	}
 	
 	@Test
 	public void testStarRegex() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("a*");
+		String regex = "a*";
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
 		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.ANY);
 		
-		Assert.assertTrue(exactQuery.equals(expectedQuery));
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
 	}
 	
 	@Test
 	public void testLiteral1() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("abc");
-		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
-		expectedQuery.operandSet.addAll(Arrays.asList(new String[]{"abc"}));
+		String regex = "abc";
 		
-		Assert.assertTrue(exactQuery.equals(expectedQuery));
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
+		GramBooleanQuery expectedAndNode = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
+		expectedAndNode.operandSet.addAll(Arrays.asList(new String[]{"abc"}));
+		expectedQuery.subQuerySet.add(expectedAndNode);
+		
+//		printTranslatorResult(regex);
+
+		Assert.assertTrue(simplifiedDNF.equals(expectedQuery));
 	}
 	
 	@Test
 	public void testLiteral2() {
-		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate("ab");
-		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.AND);
-		expectedQuery.operandSet.addAll(Arrays.asList(new String[]{}));
+		String regex = "ab";
+		
+		GramBooleanQuery exactQuery = RegexToGramQueryTranslator.translate(regex);
+		GramBooleanQuery dnf = GramBooleanQuery.toDNF(exactQuery);
+		GramBooleanQuery simplifiedDNF = GramBooleanQuery.simplifyDNF(dnf);
+		
+		GramBooleanQuery expectedQuery = new GramBooleanQuery(GramBooleanQuery.QueryOp.OR);
```

Comment:
"ab" can't form a trigram, so the result is empty. Since it's DNF, so the base node is OR. So it' an empty OR.